test/functionapp.test.ts

```diff
@@ -0,0 +1,99 @@
+import * as assert from 'assert';
+import { ResourceManagementClient } from 'azure-arm-resource';
+import { WebSiteManagementClient, WebSiteManagementModels } from 'azure-arm-website';
+import { IHookCallbackContext, ISuiteCallbackContext } from 'mocha';
+import * as vscode from 'vscode';
+import { AzureTreeDataProvider, DialogResponses, TestAzureAccount, TestUserInput } from 'vscode-azureextensionui';
+import { ext } from '../src/extensionVariables';
+import { FunctionAppProvider } from '../src/tree/FunctionAppProvider';
+import * as fsUtil from '../src/utils/fs';
+import { longRunningTestsEnabled } from './global.test';
+
+// tslint:disable-next-line:max-func-body-length
+suite('Function App actions', async function (this: ISuiteCallbackContext): Promise<void> {
+    this.timeout(1200 * 1000);
+    const resourceGroupsToDelete: string[] = [];
+    const resourceName: string = fsUtil.getRandomHexString().toLowerCase();
+    const testAccount: TestAzureAccount = new TestAzureAccount();
+    suiteSetup(async function (this: IHookCallbackContext): Promise<void> {
+        if (!longRunningTestsEnabled) {
+            this.skip();
+        }
+        this.timeout(120 * 1000);
+        await testAccount.signIn();
+        ext.tree = new AzureTreeDataProvider(FunctionAppProvider, 'azureFunctions.startTesting', undefined, testAccount);
+    });
+    suiteTeardown(async function (this: IHookCallbackContext): Promise<void> {
+        if (!longRunningTestsEnabled) {
+            this.skip();
+        }
+        this.timeout(1200 * 1000);
+        const client: ResourceManagementClient = getResourceManagementClient(testAccount);
+        for (const resourceGroup of resourceGroupsToDelete) {
+            if (await client.resourceGroups.checkExistence(resourceGroup)) {
+                console.log(`Deleting resource group "${resourceGroup}"...`);
+                await client.resourceGroups.deleteMethod(resourceGroup);
+                console.log(`Resource group "${resourceGroup}" deleted.`);
+            } else {
+                // If the test failed, the resource group might not actually exist
+                console.log(`Ignoring resource group "${resourceGroup}" because it does not exist.`);
+            }
+        }
+    });
+    test('createFunctionApp', async () => {
+        resourceGroupsToDelete.push(resourceName);
+        const testInputs: string[] = [resourceName, '$(plus) Create new resource group', resourceName, '$(plus) Create new storage account', resourceName, 'East US'];
+        ext.ui = new TestUserInput(testInputs);
+        await vscode.commands.executeCommand('azureFunctions.createFunctionApp', testAccount.getSubscriptionId());
+        const client: WebSiteManagementClient = getWebsiteManagementClient(testAccount);
+        const createdApp: WebSiteManagementModels.Site = await client.webApps.get(resourceName, resourceName);
+        assert.ok(createdApp);
+    });
+    test('stopFunctionApp', async () => {
```

Comment:
nit: Can you have a break (enter/return) between tests?  It makes the readability much better :)